app/controllers/addresses_controller.rb

```diff
@@ -0,0 +1,24 @@
+class AddressesController < ApplicationController
+  load_resource
+
+  def create
+    @address = current_user.addresses.create(address_params)
+    show_errors(:create)
+    redirect_to edit_user_registration_path
+  end
+
+  def update
+    @address.update(address_params)
+    show_errors(:update)
+    redirect_to edit_user_registration_path
+  end
+
+  private
+  def address_params
+    params.require(:address).permit(:firstname, :lastname, :address, :city, :zipcode, :country, :phone, :address_type)
+  end
+
+  def show_errors(action)
```

Comment:
This naming is confusing. Method `show_errors` shows both alerts and notices. Maybe `show_flash`?